controllers/accounts.js

```diff
@@ -44,61 +44,40 @@ const getTableActions = (item, path) => {
     return tableActions;
 };
 
-/*
-const getCreateHandler = (service) => {
-    return function (req, res, next) {
-        req.body.schoolId = req.query.schoolId;
-        api(req).post('/' + service + '/', {
-            // TODO: sanitize
-            json: req.body
-        }).then(data => {
-            if (req.body.silent !== 'on')
-                sendMailHandler(data, req);
-            res.redirect(req.header('Referer'));
-        }).catch(err => {
-            next(err);
-        });
-    };
-};
-*/
 const getUpdateHandler = (service) => {
     return function (req, res, next) {
-        /**if (req.body.roles[0].includes(',')) {
-            req.body.roles = req.body.roles[0].split(',');
-        }**/
-        api(req).patch('/' + service + '/' + req.params.id, {
-            // TODO: sanitize
-            json: req.body
-        }).then(data => {
-            res.redirect(req.header('Referer'));
-        }).catch(err => {
-            next(err);
-        });
+        const { username, password, activated } = req.body;
+        api(req, { useCallback: false, json: true, version: 'v3' })
+            .patch('/' + service + '/' + req.params.id, {
+                json: {
+                    username,
+                    password,
+                    activated: activated === 'on',
+                }
+            })
+            .then(() => res.redirect(req.header('Referer')))
+            .catch(err => next(err));
     };
 };
 
-
 const getDetailHandler = (service) => {
     return function (req, res, next) {
-        api(req).get('/' + service + '/' + req.params.id).then(data => {
-            res.json(data);
-        }).catch(err => {
-            next(err);
-        });
+        api(req, { useCallback: false, json: true, version: 'v3' })
+            .get('/' + service + '/' + req.params.id)
+            .then(data => res.json(data))
+            .catch(err => next(err));
     };
 };
 
-
 const getDeleteHandler = (service) => {
     return function (req, res, next) {
-        api(req).delete('/' + service + '/' + req.params.id).then(account => {
-                api(req).delete('/users/' + account.userId)
-                    .then(_ => {
-                        res.redirect(req.header('Referer'));
-                    });
-        }).catch(err => {
-            next(err);
-        });
+        api(req, { useCallback: false, json: true, version: 'v3' })
```

Comment:
Nested promises are difficult to read and error prone. I would like some promise chaining or even better `await ...`

In this case I am quite sure that an error while deleting the user is swallowed, because the promise is missing `.catch(...)`